runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java

```diff
@@ -211,8 +211,11 @@ public Instance initialize(boolean start) {
             // TODO: can be parallelized?
             for (int i = 0; i < this.functions.length; i++) {
                 if (this.function(i) != null) {
-                    new TypeValidator()
-                            .validate(this.function(i), this.types[this.functionType(i)], this);
+                    var funcType = this.functionType(i);
+                    if (funcType >= this.types.length) {
+                        throw new InvalidException("unknown type");
```

Comment:
Nitpick: wouldn't be better to also add in the message of the exception the id of the unknown type? This could help for debuggability. Same thing for global and elements after in this file.

Reply:
done :+1: